### DISCUSSION TITLE
"Verbindung fehlgeschlagen" bekomme keine Verbindung aus HA zu RaspberryMatic aufgebaut

### DISCUSSION THREAD

**Discussion by zilli82** [SukramJ/aiohomematic#1259] [Q&A]
Hallo,

ich hatte bisher nur eine alte CCU2 am laufen und bin jetzt umgestiegen auf RaspberryMatic im Docker Container auf einem Pi4.
Da ich nicht nur ELV Homemativ Geräte habe, möchte ich jetzt gerne auch noch HomeAssistant in einem eigenen Docker Container ans laufen bekommen und alle Geräte im HA integrieren.

Konfiguration:
Als Hardware Raspberry Pi4
Dochker
Portainer in eigenem Docker Container
Raspberry Matic in eigenem Docker Container
HomeAssistant in eigenem Docker Container

RaspberryMatic läuft
HomeAssistant läuft

In HomeAssistant habe ich über HACS Homematic(IP) Local installiert. Was mich dann schon irritiert hat ist, dass meine CCU nicht automatisch gefunden wurde. Also habe ich über die Doku den My button: ADD Integration geklickt. Dort habe ich alles so wie in der Anleitung eingetragen, aber ich bekomme immer "Verbindung fehlgeschlagen".
![grafik](https://github.com/danielperna84/hahomematic/assets/84272521/de13fa24-f606-4de6-9601-64fe34e98f46)
![grafik](https://github.com/danielperna84/hahomematic/assets/84272521/e5c77f43-9f7c-4c46-bd71-9374411fdb02)

Hier mal meine Einstellungen in RaspberryMatic:
Sicherheit -> Authentifizierung aktiv = false
Firewall:
![grafik](https://github.com/danielperna84/hahomematic/assets/84272521/18c3a322-e4b5-4936-bb9d-d19d88a3acb6)

Keine Angst soll nicht so bleiben, aber ich wollte erstmal alles offen haben um dort Fehler auszuschließen und dann stepweise wieder aktivieren.

Hier noch meine Container im Überblick
![grafik](https://github.com/danielperna84/hahomematic/assets/84272521/3e54e3ad-99ff-4136-9431-24f65fe38b8b)

Leider ist auch Docker für mich neu. Ich bin mir nicht sicher ob es an den Network-Einstellungen liegen könnte (Da habe ich auch schon iniges rumprobiert). Komisch finde ich das bei HA keine IP Adresse auftaucht. Erreichen tue ich HA aber über Host-Adresse mit Port: http://192.168.168.82:8123

Vielleicht kann mir ja hier jemand helfen.

Danke euch schonmal.

Warum mein CCU Container unhealthy ist, weiß ich auch noch nicht, das ist aber ein hoffentlich ein anderes Thema, das ich danach angehen will. Ein neuaufsetzen des Containers über das offizielle Script hatte da nicht geholfen.

**Comment by SukramJ** [comment]
> Leider ist auch Docker für mich neu. 

Ich kann leider nicht verstehen, warum Du ausgerechnet die komplizierteste Installationsmöglichkeit für HA/RM in einer Docker Umgebung nehmen musst, wenn Du eigentlich nicht so wirklich Ahnung davon hast was Du da machst.

Ich kann dir nur HAOS empfehlen, das im Kern eine Docker Installation ist, in der Du zusätzlich auch RaspberryMatic als Add-On (Container) installieren kannst. Hier gibt es auch reichlich Videos und andere Tutorials.

In Bezug auf diese Integration findest Du ein paar Hinweise in der [Doku](https://github.com/danielperna84/custom_homematic#custom_homematic) (Stichwort: Callback).

Bitte habe Verständnis dafür, das wir hier eine funktionierende CCU Umgebung voraussetzen.

**Reply by zilli82** [reply]
Hallo SukramJ,

du musst das auch nicht verstehen. Ob das die komplizierteste Installationsmöglichkeit für HM/RM ist kann ich nicht beurteilen, ist mir auch egal. Aus meiner aktuellen Sicht ist es für mich die schönste, weil ich die System schön getrennt von einander laufen habe.

Zudem möchte ich ja was lernen. Ich weiß nicht was dir in die Wiege gelegt wurde so das du alles kannst was du machst. Ich muss immer wieder neue Dinge erlernen und das kann ich am besten wenn ich diese Dinge mache.

Meiner Meinung nach funktioniert meine CCU Umgebung. Zumindest läuft alles und ich kann per APP und WebInterface drauf zugreifen
Das mit den Callbacks habe ich auch schon gelesen, komme hier aber nicht weiter. Vielleicht sehe ich auch nach Stunden der Recherche den Wald nicht mehr.

Vielleicht finde ich ja Community in der mir geholfen wird und ich lernen kann.

**Reply by danielperna84** [reply]
Das _Problem_ hierbei ist, dass wenn du eigentlich keine Ahnung davon hast, jemand dich an der Hand nehmen, und Schritt für Schritt alle möglichen Fehlerquellen betrachten muss. Das ist unglaublich aufwändig, wenn der um Hilfe bittende User gerade Neuland betritt.

Wie SukramJ schon sagt, hast du bei HAOS im Prinzip auch die separaten Container. Stell' es dir einfach wie ein automatisch eingerichtetes Docker-Setup mit allem drum und dran vor. Nur eben, dass alles direkt funktioniert. Ja, da lernst du nicht so viel dabei. Aber zum lernen wäre ein anderes Projekt, auf einem separaten System, eher zu empfehlen.

Und jetzt wo das gesagt ist: mein erster Versuch wäre `192.168.168.82` und `8123` bei Callback IP und Callback Port einzugeben. Und wenn das nicht hilft, dann in die Logs schauen und welchen Fehler die Integration hat.

**Reply by zilli82** [reply]
Hi Daniel,

ich möchte ja garnicht das mich jemand an die Hand nimmt. Fand die Reaktion von SukramJ nur sehr abweisend. Kann aber auch sein das es daran liegt das ich wenig geschlafen habe auf der Fehlersuche.

Danke dir, ich werde das ausprobieren und weiter versuchen mehr und mehr zu verstehen.

**Comment by onn0** [comment]
Hi,

dass bei deinem HA container keine IP angezeigt wird, ist kein Fehler. Du hast ja wahrscheinlich host als Netzwerk eingestellt. Dann wird keine IP angezeigt.
https://stackoverflow.com/questions/43803849/my-docker-container-does-not-have-ip-address-why

Ich bin bei deinem Beitrag gelandet, weil ich vor der gleichen Fehlermeldung stand. Die Configuration sieht bei mir etwas anders aus. Raspberrymatic läuft bei mir direkt auf einem raspberrypi. Homeassistant aber genau wie hier im Container auf einer anderen Maschine. Im HA log war auch kein wertvoller Hinweis zu finden:
2023-10-15 01:31:13.187 WARNING (MainThread) [hahomematic.central] ()

Ich habe erst versucht die optionalen Callback settings auszufüllen, ohne den gewünschten Effekt. Wenn ich jedoch unter "Konfiguration der CCU Schnittstellen" BidCos Wired nicht aktiviere funktioniert die Kopplung. Auf den ersten Blick einwandfrei. Im Readme steht ja auch, dass nur die wirklich verwendeten Protokolle aktiviert werden sollen, war mir allerdings nicht sicher ob ich nicht doch BidCos Wired brauche.

**Comment by x4N70pHyLL** [comment]
Ich reihe mich hier mal ein, habe nahezu das selbe Setup wie @onn0 : Pi3 mit HAOS und Z-Wave + Zigbee Hardware und entsprechend HA Addons dazu. Hauptinstanz des HomeAssistant läuft aber auf meinem NAS als Container, von dort verbinde ich mich mit den passenden Integrationen zum Pi3. Das ganze ist en Testsetup vor dem (physischen Wohnungs-) Umzug in die Zielumgebung. Dort habe ich dann auch HomeMatic für die Heizung und ich würde gerne statt mit einem HM Access Point zu starten die CCU / RaspberryMatic wie Zigbee und Z-Wave parallel dazu laufen lassen.

Zum Problem: Ich kann mich trotz der in der Doku hier angegebenen freizugebenden Ports nicht verbinden.
Fehlermeldung und Log-Meldung sind leider nicht aussagefähig.

So sieht die config aus. (Callback mit 192.168.2.10 und Ports 333 und 334 - das ist der Dockerhost und 2 freie, beliebig gewählte Ports - habe ich auch schon getestet).
<img width="369" alt="Bildschirmfoto 2023-11-12 um 19 57 56" src="https://github.com/danielperna84/hahomematic/assets/81616315/7555a12b-4998-4503-9359-acb74c4c6791">

Diese Ports habe ich in im Addon auf dem Pi freigegeben:
<img width="515" alt="Bildschirmfoto 2023-11-12 um 20 04 21" src="https://github.com/danielperna84/hahomematic/assets/81616315/c6474a14-651b-446a-8ba6-afca462b3fb5">
<img width="500" alt="Bildschirmfoto 2023-11-12 um 20 04 30" src="https://github.com/danielperna84/hahomematic/assets/81616315/26840f72-842b-466b-8a2f-abe3a8fb4e3c">

Und so ist die Firewall in RaspberryMatic eingestellt:
<img width="499" alt="Bildschirmfoto 2023-11-12 um 20 03 06" src="https://github.com/danielperna84/hahomematic/assets/81616315/14d1e1e6-57d9-448b-8ec0-923bed7eb221">

Eine Frage um das auszuschließen: Ich habe aktuell noch KEINE HM Hardware im Pi3 - kann der Fehler da her kommen?

Ansonsten bin ich für Hinweise dankbar.
Grüße
x4N70

**Reply by danielperna84** [reply]
> Eine Frage um das auszuschließen: Ich habe aktuell noch KEINE HM Hardware im Pi3 - kann der Fehler da her kommen?

Ja. Wurde gerade erst vor knapp 2 Stunden hier geschrieben: https://homematic-forum.de/forum/viewtopic.php?p=784610#p784612

**Reply by x4N70pHyLL** [reply]
Danke für den Link und die schnelle Antwort, da gehts ja aber um das BidCos wired, das hatte ich auch schonmal woanders gelesen, dass da die HW wirklich vorhanden sein muss. Das habe ich aber nicht angehakt (bzw. beides probiert)
Ich habe bei den Häkchen der Dienste so ziemlich alle Kombis durchprobiert, vor allem aber auch schon NUR HomeMaticIP und sonst nichts.
Ich habe auch nicht den "connection refused" Fehler, bei mir heißt es nur "fehlgeschlagen".

**Reply by danielperna84** [reply]
Es spielt keine Rolle ob Wired oder nicht. Wenn da keine Geräte hinten dran sind (was ja nicht sein kann, wenn du kein Funkmodul dran hast), dann kann der auch keine sinnvollen Antworten liefern.

Und das _Connection refused_ bekommt man nicht auf der Config-Seite angezeigt. Das ist in den Logs. Und ohne Logs kann man den Fehler auch nicht weiter eingrenzen.

**Reply by Baxxy13** [reply]
Ich präzisiere das mal ein bisschen. Die verfügbaren Interfaces (BidCos-RF, HmIP-RF usw.) werden auf der CCU nur angelegt wenn auch ein Funkmodul mit entsprechenden Kapazitäten vorhanden ist. Beim HmIP-RFUSB-TK z.B. gibt's nur HmIP-RF, der "Originale" kann alle. 

Ob da Geräte an den Interfaces hängen spielt keine Rolle. Die Integration kann sich eben auf das jeweilige (nicht vorhandene) Interface nicht verbinden. Man darf also nur anhaken was auf der Zentrale wirklich vorhanden ist. Ist gar kein Interface vorhanden (kein Funkmodul) dann kann sich die Integration auch nicht verbinden.

**Reply by x4N70pHyLL** [reply]
Danke für die etwas detailliertere Antwort. Ich habe inzwischen die Hardware bekommen und eingesteckt - es tut tatsächlich.
Ich hatte ursprünglich meine Frage hier angehängt, weil ich persönlich das sinngemäße "mach halt keinen Container wenn du dich nicht damit auskennst" oben keine passende Antwort fand. Daher möchte ich fairerweise hier auch für den nächsten, der auf google diesen Thread findet sagen "Ohne Hardware leider kein Test möglich" bzw. wird sich die Integration nicht verbinden lassen.
